### DISCUSSION TITLE
What is the right way of release a npm components packages that uses Tailwind?

### DISCUSSION THREAD

**Discussion by getJv** [tailwindlabs/tailwindcss#13266] [General]
Hey Hello.
I published a package let's call it **pkg-a**.  So I installed it into a fresh project ( **proj-a** ) with tailwind ready to use.
BUT besides **proj-a** already uses tailwind it does not render the styles used by  **pkg-a**.

[Checking the documents][] I learned if we map the package into the `content key` of tailwind.config.js it will work. and it does:

```javascript
// tailwind.config.js of proj-a
module.exports = {
  content: [
      './src/**/*.{html,js}',
+    './node_modules/@pkg-a/dist/**/*.js',
  ],
  // ...
}

```

But even if this solves the problem, it does not seem right... if I recommend it to everyone and then install my tailwind packages they will end with several entries in their `content array` 

```javascript
// tailwind.config.js of proj-a
module.exports = {
  content: [
    './src/**/*.{html,js}',
    './node_modules/@pkg-a/dist/**/*.js',
+    './node_modules/@pkg-b/dist/**/*.js',
+   './node_modules/@pkg-c/dist/**/*.js',
+   './node_modules/@pkg-d/dist/**/*.js',
  ],
  // ...
}

```

So,  what are the best practices for releasing packages that use tailwind?

[Checking the documents]: https://tailwindcss.com/docs/content-configuration#working-with-third-party-libraries

**Comment by adamwathan** [comment]
You have it right, that's the recommended way to do it 👍  You have to tell Tailwind about any folders it needs to scan for classes.

**Comment by Vijaykv5** [comment]
@getJv  Yes! that's a right way to release a package that uses tailwind.

**Comment by getJv** [comment]
Thanks for the fast answer @adamwathan and @Vijaykv5, this confirmation helped me a lot. Im going to add this step to the lib docs.

Great week for you guys.
